### DISCUSSION TITLE
Segmentation Fault when using graph_substitute for MX function

### DISCUSSION THREAD

**Discussion by agoshtasbi** [casadi/casadi#3856] [Q&A]
Using casadi v3.6.6, I get segmentation fault running this code:

`import casadi as ca

def find_func_call(expr):
    f_call = []
    if expr.n_dep() > 0:
        if expr.is_op(ca.OP_CALL):
            f_call += [expr]
        else:
            for i in range(expr.n_dep()):
                f_call += find_func_call(expr.dep(i))
    else:
        if expr.is_op(ca.OP_CALL):
            f_call += [expr]
    return f_call

x = ca.MX.sym('x', 2)
z = ca.MX.sym('z', 1)
g1 = ca.Function('g', [z], [2.1 * z])
g2 = g1(x[0])
fnc_call = find_func_call(g2)
print(ca.graph_substitute(fnc_call, [g2], [ca.MX(1)]))`

Running this returns:

> Process finished with exit code 139 (interrupted by signal 11:SIGSEGV)

I am trying to write some code to replace a function node in an expression graph with some other expression. Is there a way to achieve this without memory issues?

**Comment by jgillis** [comment]
Simplified
```python
from casadi import *

x = MX.sym('x', 2)
z = MX.sym('z', 1)
g1 = Function('g', [z], [2.1 * z])
g2 = g1(x[0])
fnc_call = g2.dep(0)
print(graph_substitute(fnc_call, [g2], [1]))

```

**Comment by jgillis** [comment]
Thanks for catching this, @agoshtasbi 
Fix is pending.

Per our rules, you have earned one free beer (or equivalent) when you bump into @jgillis or @jaeandersson
